src/hooks/useAuthorities/useAuthorities.js

```diff
@@ -0,0 +1,63 @@
+import { useQuery } from 'react-query';
+import queryString from 'query-string';
+
+import {
+  useOkapiKy,
+  useNamespace,
+} from '@folio/stripes/core';
+
+import { template } from 'lodash';
+
+import { buildQuery } from '../utils';
+
+const AUTHORITIES_API = 'search/authorities';
+
+const useAuthorities = ({ searchQuery, searchIndex, location, history }) => {
+  const ky = useOkapiKy();
+  const [namespace] = useNamespace();
+
+  const queryParams = {
+    query: searchQuery,
+  };
+
+  const compileQuery = template(
+    buildQuery(searchIndex),
+    { interpolate: /%{([\s\S]+?)}/g },
+  );
+
+  const cqlQuery = queryParams.query?.trim().replace('*', '').split(/\s+/)
+    .map(query => compileQuery({ query }))
+    .join(' and ');
+
+  const searchParams = {
+    query: cqlQuery,
+  };
+
+  const { isFetching, data } = useQuery(
+    [namespace, searchParams],
+    async () => {
+      if (!searchQuery) {
+        return { authorities: [], totalRecords: 0 };
+      }
+
+      const locationSearchParams = queryString.parse(location.search);
+      locationSearchParams.query = searchQuery;
+      locationSearchParams.qindex = searchIndex;
+      location.search = queryString.stringify(locationSearchParams);
```

Comment:
Yes, we need this line, because on the line 43 here we parse a query string into an object, then on lines 44-45 replace params with new values and finally on line 46 we stringify an object back into a query string to use it in the `history.replace`. Without this line it just doesn't work (e.g. when initially no search params are applied and `location.search` is empty if then we search by some query and/or index the url would not be updated because `location.search` passed into `history.replace` would be still empty).